plugins/logs/plugin.go

```diff
@@ -668,6 +671,11 @@ func (p *Plugin) reconfigure(config interface{}) {
 func (p *Plugin) encodeAndBufferEvent(event EventV1) {
 	if p.limiter != nil {
 		if !p.limiter.Allow() {
+			plugin := status.Lookup(p.manager)
```

Comment:
Might be I'm missing some background context here, but It seems a little odd to me that the decision logger plugin would write metrics directly to another plugin (the status one). This couples the collection of metrics to the reporting of said metrics, no? Seems like this would be something one might expect to find in eg. Prometheus. 

Tried digging in to some code around this but it only increased my confusion 😆 From what I can see the metrics object in the status plugin is in turn only ever set by the _discovery_ plugin.. so there seems to be a whole chain of (from what I can tell, undocumented) events that need to fulfilled here in order for this to happen.

Reply:
Refactored the code to pass the metrics to the decision logger when it's created. The [runtime](https://github.com/open-policy-agent/opa/blob/master/runtime/runtime.go#L260) package instantiates the discovery plugin and also sets the metrics object on the plugin. So the metrics will be set on the status and decision log plugin both in the discovery and non-discovery scenarios.